### PR TITLE
Added New Features & Fixes

### PR DESCRIPTION
**Pull request by DontEatOreo**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -70,11 +71,41 @@
 };
 downloadPath.SetDefaultValue(Path.Combine(Environment.CurrentDirectory, "cunnycli-downloads"));
 
-Option<string[]?> excludeTags = new(new[] { "--exclude-tags", "-et" }, "Tags to exclude, separated by space")
+Option<string?> excludeTags = new(new[] { "--exclude-tags", "-et" }, "Tags to exclude, separated by space")
```
Comment (GermanBread) [inline Program.cs:74]: Why is this not a list anymore?

File: Program.cs [hunk 2]
```diff
@@ -1,178 +1,264 @@
...
+             outputJson,
+             skip,
+             cunnyApiUrl,
+             excludeTags,
+             proxyOption,
+             downloadPath
+         })
+    rootCommand.AddOption(option);
+
+var progress = 0;
```
Comment (GermanBread) [inline Program.cs:159]: Can you change this to 1?

```suggestion
var progress = 1;
```

File: Program.cs [hunk 3]
```diff
@@ -1,178 +1,264 @@
...
+            Directory.CreateDirectory(dirPath);
+
+        Globals.Logs.Add($"Downloading \u001b[22;37m{Path.GetFileName(filePath)}\u001b[0m");
+        using var downloadStream = Globals.Client.GetStreamAsync(itemValue.ImageURL).Result;
+        using var writeStream = File.Open(filePathPart, FileMode.Create);
+        downloadStream.CopyTo(writeStream);
+        File.Move(filePathPart, filePath);
+        Globals.Logs.Add($"Saved \u001b[22;37m{Path.GetFileName(filePath)}\u001b[0m");
+        progress++;
+    });
```
Comment (GermanBread) [inline Program.cs:219]: Can you add this QoL feature?

```suggestion
    });

    Console.WriteLine($"\u001b[u\u001b[0JSaved to \u001b[22;37m{downloadPathValue}\u001b[0m");
```

File: Program.cs [hunk 4]
```diff
@@ -1,178 +1,264 @@
...
-    private static readonly List<Task> downloadThreads = new();
-}
\ No newline at end of file
+
+    var results = await CunnyApiClient.Get(cunnyApiUrlValue, booruValue, tagsValue, amountValue, skipValue);
+    if (results is null)
+        return;
+
+    if (excludeTagsValue is not null)
+        results = new List<CunnyJsonElement>(results.Where(x => !x.Tags.Any(excludeTagsValue.Contains)));
```
Comment (GermanBread) [inline Program.cs:190]: Unnecessary allocation of a new list?

File: Program.cs [hunk 5]
```diff
@@ -1,178 +1,264 @@
...
+    val.ErrorMessage = "Invalid Cuuny API URL";
+});
+cunnyApiUrl.SetDefaultValueFactory(Globals.DefaultCunnyApiurl.ToString);
+
+Option<string> downloadPath = new(new []{"--path", "-p"}, "A valid path to a directory") {
+    IsRequired = true
+};
+downloadPath.SetDefaultValue(Path.Combine(Environment.CurrentDirectory, "cunnycli-downloads"));
+
+Option<string?> excludeTags = new(new[] { "--exclude-tags", "-et" }, "Tags to exclude, separated by space")
```
Comment (GermanBread) [inline Program.cs]:
```suggestion
Option<string[]?> excludeTags = new(new[] { "--exclude-tags", "-et" }, "Tags to exclude, separated by space")
```

**Review by GermanBread — DISMISSED** [review]

File: Program.cs [hunk 6]
```diff
@@ -190,7 +190,7 @@ async Task DownloadHandler(InvocationContext invocationContext)
         results = new List<CunnyJsonElement>(results.Where(x => !x.Tags.Any(excludeTagsValue.Contains)));
 
     Globals.Logs.CollectionChanged += (_, _) =>
-        Console.WriteLine(
+        Console.Write($"\u001b[u\u001b[s\u001b[22;37m[\u001b[32m{progress}\u001b[37m/\u001b[0m{amountValue}\u001b[37m]\u001b[0m {Globals.Logs[^1]}\u001b[0J");
             $"\u001b[s\u001b[22;37m[\u001b[32m{progress}\u001b[37m/\u001b[0m{amountValue}\u001b[37m]\u001b[0m {Globals.Logs[^1]}\u001b[0J\u001b[u");
```
Comment (GermanBread) [inline Program.cs:194]: You need to remove this line too

File: Program.cs (hunk 4 above)
Comment (DontEatOreo) [reply Program.cs:190]: What suggestions do you have besides making new list to store the elements with the exclude tags removed?

Comment (GermanBread) [reply Program.cs:190]: Since IEnumerable<T> is evaluated lazy it is better to create a List<T> here. For now.

File: Program.cs (hunk 6 above)
Comment (GermanBread) [reply Program.cs]:
```suggestion
```

**Review by GermanBread — APPROVED** [review]
LGTM

**Pull request merged**